presto-hive/src/main/java/com/facebook/presto/hive/HiveConnector.java

```diff
@@ -70,6 +71,7 @@
     private final ClassLoader classLoader;
     private final ConnectorPlanOptimizerProvider planOptimizerProvider;
     private final ConnectorMetadataUpdaterProvider metadataUpdaterProvider;
+    private final ConnectorMetadataUpdateHandleSerdeProvider metadataUpdateHandleSerdeProvider;
```

Comment:
[major] is `MetadataUpdate` the only object we need to worry or there are other handles we need to handle in the followup PRs? Because if `MetadataUpdate` is the only one, we don't have to build a `metadataUpdaterProvider` + `metadataUpdateHandleSerdeProvider` in separate path. Those two can be combined into one. As long as we switch interface `ConnectorMetadataUpdateHandle` to a concrete class with `byte[]` as the value, we should be good.

Reply:
`MetadataUpdate ` is not the only one. We will have to support `ConnectorSplit` and `ConnectorTransactionHandle` as a part of `TaskUpdateRequest` conversion.

Reply:
refactored the code to go with a single `ConnectorTypeSerdeProvider` instead of providers for each type.